torch/_inductor/codegen/cpp.py

```diff
@@ -1828,7 +1891,61 @@ def gen_loop(loop: LoopLevel, in_reduction=False):
                         gen_kernel(kernels[0])
 
             stack.enter_context(code.indent())
-            if loop_nest.root:
+            if outer_loop_fusion:
```

Comment:
I think this is related to https://github.com/pytorch/pytorch/pull/121064#issuecomment-1985489574, and https://github.com/pytorch/pytorch/pull/121625#pullrequestreview-1927398754, so I'm not sure whether it can be fixed without a large refactor.

Reply:
> I think this is related to [#121064 (comment)](https://github.com/pytorch/pytorch/pull/121064#issuecomment-1985489574), and [#121625 (review)](https://github.com/pytorch/pytorch/pull/121625#pullrequestreview-1927398754), so I'm not sure whether it can be fixed without a large refactor.

@leslie-fang-intel and I talked offline and think that it should be doable without a lot of refactoring, by reusing the existing `gen_loops` and `gen_kernel` based on the new `OuterLoopFusedKernel` abstraction. Let's see how it turns out.

Reply:
`OuterLoopFusedKernel` abstraction has been added. I will do some further code clean up.